File: kevoree-core/org.kevoree.core/src/main/java/org/kevoree/core/deploy/KevoreeDeployPhase.java
package org.kevoree.core.deploy;

import java.util.ArrayList;
import java.util.List;

import org.kevoree.api.PrimitiveCommand;
import org.kevoree.core.KevoreeCoreBean;

public abstract class KevoreeDeployPhase {

	public KevoreeDeployPhase successor = null;
	protected KevoreeCoreBean originCore;
	protected List<PrimitiveCommand> primitives = new ArrayList<PrimitiveCommand>();
	protected long maxTimeout = 30000l;
	protected boolean rollbackPerformed = false;
	
	public KevoreeDeployPhase(KevoreeCoreBean core) {
		this.originCore = core;
	}
	
    public void setMaxTime(long timeout) {
    	maxTimeout = Math.max(maxTimeout, timeout);
    }

    public void populate(PrimitiveCommand cmd) {
    	primitives.add(cmd);
    	rollbackPerformed = true;
    }
	
	public abstract boolean execute();
	public abstract void rollback();	
}

File: kevoree-core/org.kevoree.core/src/main/java/org/kevoree/core/impl/deploy/KevoreeDeployPhase.kt
package org.kevoree.core.impl.deploy

import org.kevoree.api.PrimitiveCommand

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 04/12/2013
 * Time: 09:26
 */

trait KevoreeDeployPhase {
    fun rollBack()
    fun runPhase(): Boolean
    fun populate(cmd: PrimitiveCommand)
    var sucessor: KevoreeParDeployPhase?
}
